### PR TITLE
docs(routing): add nested example

### PR DESCRIPTION
**Pull request by geromegrignon**
### 🔗 Linked issue

#2415 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Nesting routes with file based routing and using `index` files might be missing from an user's perspective.
Id adds a more complete example of a structure.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by geromegrignon** [comment]
Partially unrelated, i'm unsure about this note:
image

Does it mean the folder structure for standalone nitro is `/api` and `/routes`, while prefixed by `server`for Nuxt apps?
Because by creating a standalone nitro app nowadays, there is a root server folder too.

**Comment by Atinux** [comment]
> Does it mean the folder structure for standalone nitro is /api and /routes, while prefixed by serverfor Nuxt apps?
> Because by creating a standalone nitro app nowadays, there is a root server folder too.

Nice catch, I think we can now remove this alert in the docs.

File: docs/1.guide/2.routing.md [hunk 1]
```diff
@@ -27,6 +27,20 @@ routes/
...
+```md
+routes/
+  communities/
+    index.get.ts
+    index.post.ts
+    [id]/
+        index.get.ts
+        index.post.ts
+  hello.get.ts
+  hello.post.ts
```
Comment (Atinux) [inline docs/1.guide/2.routing.md:41]:
```suggestion
server/
  routes/
    communities/
      index.get.ts
      index.post.ts
      [id]/
          index.get.ts
          index.post.ts
    hello.get.ts
    hello.post.ts
```

Comment (Atinux) [reply docs/1.guide/2.routing.md:41]: I think it would be nice to update all examples to include `server/` for consistency.

**Review by pi0 — APPROVED** [review]
Thanks.

**Pull request merged**